File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/ahn-product-1-0/BaseInformation.cpp
// BaseInformation.cpp 
#include "BaseInformation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

BaseInformation::BaseInformation(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //baseIndexesNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* baseIndexesNode = xmlNode->FirstChildElement("baseIndexes");

   if(baseIndexesNode){baseIndexesIsNull_ = false;}
   else{baseIndexesIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- baseIndexesNode , address : " << baseIndexesNode << std::endl;
   #endif
   if(baseIndexesNode)
   {
      if(baseIndexesNode->Attribute("href") || baseIndexesNode->Attribute("id"))
      {
          if(baseIndexesNode->Attribute("id"))
          {
             baseIndexesIDRef_ = baseIndexesNode->Attribute("id");
             baseIndexes_ = boost::shared_ptr<BaseIndexes>(new BaseIndexes(baseIndexesNode));
             baseIndexes_->setID(baseIndexesIDRef_);
             IDManager::instance().setID(baseIndexesIDRef_,baseIndexes_);
          }
          else if(baseIndexesNode->Attribute("href")) { baseIndexesIDRef_ = baseIndexesNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { baseIndexes_ = boost::shared_ptr<BaseIndexes>(new BaseIndexes(baseIndexesNode));}
   }

   //conventionSetNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* conventionSetNode = xmlNode->FirstChildElement("conventionSet");

   if(conventionSetNode){conventionSetIsNull_ = false;}
   else{conventionSetIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- conventionSetNode , address : " << conventionSetNode << std::endl;
   #endif
   if(conventionSetNode)
   {
      if(conventionSetNode->Attribute("href") || conventionSetNode->Attribute("id"))
      {
          if(conventionSetNode->Attribute("id"))
          {
             conventionSetIDRef_ = conventionSetNode->Attribute("id");
             conventionSet_ = boost::shared_ptr<ConventionSet>(new ConventionSet(conventionSetNode));
             conventionSet_->setID(conventionSetIDRef_);
             IDManager::instance().setID(conventionSetIDRef_,conventionSet_);
          }
          else if(conventionSetNode->Attribute("href")) { conventionSetIDRef_ = conventionSetNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { conventionSet_ = boost::shared_ptr<ConventionSet>(new ConventionSet(conventionSetNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<BaseIndexes> BaseInformation::getBaseIndexes()
{
   if(!this->baseIndexesIsNull_){
        if(baseIndexesIDRef_ != ""){
             return boost::shared_static_cast<BaseIndexes>(IDManager::instance().getID(baseIndexesIDRef_));
        }else{
             return this->baseIndexes_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<BaseIndexes>();
   }
}
boost::shared_ptr<ConventionSet> BaseInformation::getConventionSet()
{
   if(!this->conventionSetIsNull_){
        if(conventionSetIDRef_ != ""){
             return boost::shared_static_cast<ConventionSet>(IDManager::instance().getID(conventionSetIDRef_));
        }else{
             return this->conventionSet_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ConventionSet>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/ahn-product-1-0/BaseInformation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class BaseInformation : ISerialized
    {
        public BaseInformation(XmlNode xmlNode)
        {
            XmlNode baseIndexesNode = xmlNode.SelectSingleNode("baseIndexes");
            
            if (baseIndexesNode != null)
            {
                if (baseIndexesNode.Attributes["href"] != null || baseIndexesNode.Attributes["id"] != null) 
                {
                    if (baseIndexesNode.Attributes["id"] != null) 
                    {
                        baseIndexesIDRef_ = baseIndexesNode.Attributes["id"].Value;
                        BaseIndexes ob = new BaseIndexes(baseIndexesNode);
                        IDManager.SetID(baseIndexesIDRef_, ob);
                    }
                    else if (baseIndexesNode.Attributes["href"] != null)
                    {
                        baseIndexesIDRef_ = baseIndexesNode.Attributes["href"].Value;
                    }
                    else
                    {
                        baseIndexes_ = new BaseIndexes(baseIndexesNode);
                    }
                }
                else
                {
                    baseIndexes_ = new BaseIndexes(baseIndexesNode);
                }
            }
            
        
            XmlNode initialFixedValueNode = xmlNode.SelectSingleNode("initialFixedValue");
            
            if (initialFixedValueNode != null)
            {
                if (initialFixedValueNode.Attributes["href"] != null || initialFixedValueNode.Attributes["id"] != null) 
                {
                    if (initialFixedValueNode.Attributes["id"] != null) 
                    {
                        initialFixedValueIDRef_ = initialFixedValueNode.Attributes["id"].Value;
                        InitialFixedValue ob = new InitialFixedValue(initialFixedValueNode);
                        IDManager.SetID(initialFixedValueIDRef_, ob);
                    }
                    else if (initialFixedValueNode.Attributes["href"] != null)
                    {
                        initialFixedValueIDRef_ = initialFixedValueNode.Attributes["href"].Value;
                    }
                    else
                    {
                        initialFixedValue_ = new InitialFixedValue(initialFixedValueNode);
                    }
                }
                else
                {
                    initialFixedValue_ = new InitialFixedValue(initialFixedValueNode);
                }
            }
            
        
            XmlNode conventionSetNode = xmlNode.SelectSingleNode("conventionSet");
            
            if (conventionSetNode != null)
            {
                if (conventionSetNode.Attributes["href"] != null || conventionSetNode.Attributes["id"] != null) 
                {
                    if (conventionSetNode.Attributes["id"] != null) 
                    {
                        conventionSetIDRef_ = conventionSetNode.Attributes["id"].Value;
                        ConventionSet ob = new ConventionSet(conventionSetNode);
                        IDManager.SetID(conventionSetIDRef_, ob);
                    }
                    else if (conventionSetNode.Attributes["href"] != null)
                    {
                        conventionSetIDRef_ = conventionSetNode.Attributes["href"].Value;
                    }
                    else
                    {
                        conventionSet_ = new ConventionSet(conventionSetNode);
                    }
                }
                else
                {
                    conventionSet_ = new ConventionSet(conventionSetNode);
                }
            }
            
        
        }
        
    
        #region BaseIndexes_
        private BaseIndexes baseIndexes_;
        public BaseIndexes BaseIndexes_
        {
            get
            {
                if (this.baseIndexes_ != null)
                {
                    return this.baseIndexes_; 
                }
                else if (this.baseIndexesIDRef_ != null)
                {
                    baseIndexes_ = IDManager.getID(baseIndexesIDRef_) as BaseIndexes;
                    return this.baseIndexes_; 
                }
                else
                {
                      return this.baseIndexes_; 
                }
            }
            set
            {
                if (this.baseIndexes_ != value)
                {
                    this.baseIndexes_ = value;
                }
            }
        }
        #endregion
        
        public string baseIndexesIDRef_ { get; set; }
        #region InitialFixedValue_
        private InitialFixedValue initialFixedValue_;
        public InitialFixedValue InitialFixedValue_
        {
            get
            {
                if (this.initialFixedValue_ != null)
                {
                    return this.initialFixedValue_; 
                }
                else if (this.initialFixedValueIDRef_ != null)
                {
                    initialFixedValue_ = IDManager.getID(initialFixedValueIDRef_) as InitialFixedValue;
                    return this.initialFixedValue_; 
                }
                else
                {
                      return this.initialFixedValue_; 
                }
            }
            set
            {
                if (this.initialFixedValue_ != value)
                {
                    this.initialFixedValue_ = value;
                }
            }
        }
        #endregion
        
        public string initialFixedValueIDRef_ { get; set; }
        #region ConventionSet_
        private ConventionSet conventionSet_;
        public ConventionSet ConventionSet_
        {
            get
            {
                if (this.conventionSet_ != null)
                {
                    return this.conventionSet_; 
                }
                else if (this.conventionSetIDRef_ != null)
                {
                    conventionSet_ = IDManager.getID(conventionSetIDRef_) as ConventionSet;
                    return this.conventionSet_; 
                }
                else
                {
                      return this.conventionSet_; 
                }
            }
            set
            {
                if (this.conventionSet_ != value)
                {
                    this.conventionSet_ = value;
                }
            }
        }
        #endregion
        
        public string conventionSetIDRef_ { get; set; }
        
    
        
    
    }
    
}

